packages/preview/clean-uoft-thesis/0.1.0/uoft.typ

```diff
@@ -0,0 +1,390 @@
+// - The author of the thesis.
+//
+// department:
+// - The department for which the thesis is submitted.
+//
+// main_margin_style: 
+// - "left_metric": (default) Applies 32 mm left margin and 20 mm top,
+//   right, and bottom margin to the main text.
+// - "left_imperial": Applies 1.25 in left margin and 0.75 in top, right,
+//   and bottom margin to the main text.
+// - "metric": Applies 20 mm margin to all sides of the main text.
+// - "imperial": Applies 0.75 in margin to all sides of the main text.
+#let uoft(title: none,
+          author: [*missing_param_author*],
+          department: [*missing_param_department*],
+          degree: [*missing_param_degree*],
+          graduation_year: [*missing_param_year*],
+          abstract: [],
+          acknowledgements: [],
+          body: [],
+          show_acknowledgements: true,
+          show_list_of_tables: true,
+          show_list_of_plates: false,
+          show_list_of_figures: true,
+          show_list_of_appendices: true,
+          title_page_top_margin: 5cm,
+          title_page_gap_1_height: 4cm,
+          title_page_gap_2_height: 4cm,
+          title_page_gap_3_height: 5cm,
+          title_page_gap_4_height: 3cm,
+          title_page_bottom_margin: 3cm,
+          page_size_style: "metric",
+          main_margin_style: "left_metric",
+          font_size: 12pt,
+          doc) = {
+  let (page_width, page_height) = parse_page_dims(
+    page_size_style
+  )
+  let (margin_t, margin_b, margin_l, margin_r) = parse_margin_dims(
+    main_margin_style
+  )
+  set page(
+    width: page_width,
+    height: page_height,
+    margin: (top: margin_t, bottom: margin_b, left: margin_l, right: margin_r),
+  )
+  let font_size = parse_font(font_size)
+  set text(size: font_size)
+
+  show heading: it => block(width: 100%)[
+    #set text(weight: "regular")
+    #(it)
+  ]
+
+  check_valid_degree(degree)
+
+  init_title_page(
+    title,
+    author,
+    department,
+    degree,
+    graduation_year,
+    title_page_top_margin,
+    title_page_gap_1_height,
+    title_page_gap_2_height,
+    title_page_gap_3_height,
+    title_page_gap_4_height,
+    title_page_bottom_margin,
+    page_size_style,
+  )
+
+  pagebreak()
+
+  set page(numbering: "i")
+
+  init_abstract(
+    abstract,
+    title,
+    author,
+    degree,
+    department,
+    graduation_year
+  )
+
+  set par(leading: 0.75em)
+
+  pagebreak()
+
+  if (show_acknowledgements) {
+    init_acknowledgements(acknowledgements)
+    pagebreak()
+  }
+
+  init_table_of_contents()
+
+  pagebreak()
+
+  if (show_list_of_tables) {
+    init_list_of_tables()
+    pagebreak()
+  }
+
+  if (show_list_of_plates) {
+    init_list_of_plates()
+    pagebreak()
+  }
+
+  if (show_list_of_figures) {
+    init_list_of_figures()
+    pagebreak()
+  }
+
+  if (show_list_of_appendices) {
+    init_list_of_appendices()
+    pagebreak()
+  }
+
+  set page(numbering: "1")
+  counter(page).update(1)
+
+  init_body(body)
+
+  doc
```

Comment:
I think these two arguments should be merged in one. Preferably `doc`, as it is a more common pattern to have the content of the document coming after the show rule being the main body, rather than passing it as an argument.